RecoPPS/ProtonReconstruction/plugins/CTPPSProtonProducer.cc

```diff
@@ -298,32 +330,35 @@ void CTPPSProtonProducer::produce(edm::Event &iEvent, const edm::EventSetup &iSe
 
         // do multi-RP reco if chosen
         if (doMultiRPReconstruction_ && rpIds.size() == 2) {
-          // find matching track pairs from different tracking RPs
+          // find matching track pairs from different tracking RPs, ordered: i=near, j=far RP
           std::vector<std::pair<unsigned int, unsigned int>> idx_pairs;
           std::map<unsigned int, unsigned int> idx_pair_multiplicity;
           for (const auto &i : indices) {
             for (const auto &j : indices) {
-              if (j <= i)
-                continue;
-
               const auto &tr_i = hTracks->at(i);
               const auto &tr_j = hTracks->at(j);
 
+              const double z_i = hGeometry->rpTranslation(tr_i.rpId()).z();
+              const double z_j = hGeometry->rpTranslation(tr_j.rpId()).z();
+
               const auto &pr_i = singleRPResultsIndexed[i];
               const auto &pr_j = singleRPResultsIndexed[j];
 
               if (tr_i.rpId() == tr_j.rpId())
                 continue;
 
+              if (std::abs(z_i) >= std::abs(z_j))
+                continue;
+
               bool matching = true;
 
-              if (ac.x_cut_apply && std::abs(tr_i.x() - tr_j.x()) > ac.x_cut_value)
+              if (ac.x_cut_apply && std::abs(tr_i.x() - tr_j.x() - ac.x_cut_mean) > ac.x_cut_value)
                 matching = false;
-              if (ac.y_cut_apply && std::abs(tr_i.y() - tr_j.y()) > ac.y_cut_value)
+              if (ac.y_cut_apply && std::abs(tr_i.y() - tr_j.y() - ac.y_cut_mean) > ac.y_cut_value)
```

Comment:
Sure, in principle no problem. I don't know that the time line is, but do you wish to restart the tests/signatures due to this (according to me minor) issue?

Reply:
Yes, I agree, it is a minor issue: only if there are other updates to be implemented, otherwise it can be postponed
(tests would take half day, though)

Reply:
Thanks! We (PPS) plan no further modifications. The PR is already signed by DQM, it seems that your/reconstruction signature is the last one. Then, if I remember well, the release is supposed to be built today. Do we expect further comments/updates?

(If there is enough time left, no problem for me to push the small fix later today)

Reply:
According to Silvio, submissions for pre8 will be cut sometime tomorrow. If  you can update now, I can start the tests at once, and have it signed this evening before going to bed.
Just let me know.

Reply:
OK, I will do it in ~1 h (once the Proton POG meeting is over).

Reply:
Done in 9277e16.

Below a plot showing that there is a no difference in results between
  * 9277e1614ec (red dashed)
  * 3371411d28c (green)

--> as expected.

![make_cmp](https://user-images.githubusercontent.com/17830858/82343612-5fb4ab80-99f3-11ea-8a2f-7d06da86c7fc.png)